winsup/cygwin/uinfo.cc

```diff
@@ -543,6 +543,36 @@ cygheap_user::env_systemroot (const char *name, size_t namelen)
   return psystemroot;
 }
 
+const char *
+cygheap_user::env_tty_handles(const char *name, size_t namelen)
+{
+  static unsigned initialized;
+  static char buffer[128];
+
+  if (!initialized)
+    {
+      int fd;
+      char *p = buffer;
+      strcpy(p, " ");
+      p += strlen(p);
+      for (fd = 0; fd < 3; fd++)
+        if (isatty(fd))
+          {
+            cygheap_fdget cfd (fd);
+
+            if (cfd < 0 || cfd->close_on_exec ())
+              continue;
+
+            sprintf(p, "%ld ", (unsigned long)
+              (fd ?  cfd->get_output_handle() : cfd->get_handle()));
+            p += strlen(p);
```

Comment:
Yeah, you're right. I was concerned about `sprintf()` returning `-1` in the error case, but that fear was probably too paranoid.